### PR TITLE
Mod: Launching a suitable ajax handler for checking email before POST

### PR DESCRIPTION
**Pull request by ArtemAnoshin**

### CODE REVIEW

File: js/src/apbct-public.js [hunk 1]
```diff
@@ -71,18 +71,54 @@
...
+								ctSetCookie('ct_checked_emails', JSON.stringify(ctCheckedEmails));
+							}
+						},
+					}
+				);
+			} else if( ctPublicFunctions.data__ajax_type === 'admin_ajax' ) {
+				apbct_public_sendAJAX(
+					{
+						action: 'apbct_email_check_before_post',
+						data : {'email' : current_email},
```
Comment (safronik) [inline js/src/apbct-public.js:110]: We should use `email : current_email` here, instead of object

File: inc/cleantalk-common.php [hunk 2]
```diff
@@ -593,6 +594,29 @@ function apbct_js_keys__get__ajax()
...
+ */
+function apbct_email_check_before_post()
+{
+    global $apbct;
+    $ajax_type = $apbct->data['ajax_type'];
+    $email = isset($_POST['data']['email']) ? trim($_POST['data']['email']) : null;
+
+    if ($ajax_type === 'rest') {
+        $email = trim(Post::get('email'));
+    }
```
Comment (safronik) [inline inc/cleantalk-common.php:608]: Code above can be replaced to `$email = trim(Post::get('email'));`

**Review by safronik — CHANGES_REQUESTED** [review]

File: js/src/apbct-public.js (hunk 1 above)
Comment (ArtemAnoshin) [reply js/src/apbct-public.js]: Done

File: inc/cleantalk-common.php (hunk 2 above)
Comment (ArtemAnoshin) [reply inc/cleantalk-common.php]: Done

**Review by safronik — APPROVED** [review]

**Pull request merged**